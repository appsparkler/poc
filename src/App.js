
// import all required libraries
import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom'
import 'materialize-css/dist/css/materialize.min.css'
import 'materialize-css/dist/js/materialize.min.js'
//
import Navbar from './components/Navbar'
//
import Home from './components/Home'
import About from './components/About'
import Contact from './components/Contact'
import Post from './components/Post'

function App() {
  return (
    <BrowserRouter>
      <div className='App'>
        <Navbar />
        <Route exact path="/" component={Home} />
        <Route path="/contact" component={Contact}/>
        <Route path="/about" component={About} />
        <Route path="/:post_id" component={Post} />
      </div>
    </BrowserRouter>
  )
}

export default App
