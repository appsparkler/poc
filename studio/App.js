import React from 'react'
import SignUpForm from './components/SignUpForm'

export default () => {
  return (
    <div>
      <h1>Studio</h1>
      <SignUpForm />
    </div>
  )
}
