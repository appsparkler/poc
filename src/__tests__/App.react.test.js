import React from 'react'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import App from '../App.react'

it('should render the expected HTML', () => {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    project: {
      projects: [{
        id: '1',
        title: 'Title 1',
        content: 'Content...',
      }],
    },
  })
  const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
