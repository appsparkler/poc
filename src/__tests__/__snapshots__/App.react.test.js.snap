// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected HTML 1`] = `
<div
  className="App"
>
  <nav>
    <div
      className="nav-wrapper grey darken-3"
    >
      <div
        className="container"
      >
        <a
          className="brand-logo"
          href="/"
          onClick={[Function]}
        >
          MarioPlan
        </a>
        <ul
          className="right"
        >
          <li>
            <a
              href="/create"
              onClick={[Function]}
            >
              Create Project
            </a>
          </li>
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Logout
            </a>
          </li>
          <li>
            <a
              className="btn btn-floating pink lighten-3"
              href="/"
              onClick={[Function]}
            >
              NN
            </a>
          </li>
        </ul>
        <ul
          className="right"
        >
          <li>
            <a
              href="/signin"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li>
            <a
              href="/signup"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="dashboard container"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m6"
      >
        <div
          className="project-list section"
          style={
            Object {
              "maxHeight": "500px",
              "overflow": "auto",
              "padding": "0 20px",
            }
          }
        >
          <a
            href="/project/1"
            onClick={[Function]}
          >
            <div>
              <div
                className="project-summary card"
              >
                <div
                  className="card-content grey-text text-darken-3"
                >
                  <span
                    className="card-title"
                  >
                    Title 1
                  </span>
                  <p>
                    Posted By Net Ninja
                  </p>
                  <p
                    className="grey-text"
                  >
                    3rd September, 2am
                  </p>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        className="col s12 m5 offset-m1"
      >
        <div>
          <p>
            Notifications
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
